### ISSUE TITLE
[Exception][PyCharm 2022.2.2] com.intellij.diagnostic.PluginException: commitDocument() left PSI inconsistent: not co...

### ISSUE THREAD

**Issue by marcinwrobel1986** [ArtsiomCh/CMake#36]
# User Description
_User did not enter any detailed information._

# Stack Trace
```
com.intellij.diagnostic.PluginException: commitDocument() left PSI inconsistent: not committed document DocumentImpl[file:///home/marcin/Dokumenty/NGA/projects/pressio-tutorials/end-to-end-roms/tests/CMakeLists.txt], File[CMake File CMakeLists.txt, Language: CMake, com.intellij.psi.SingleRootFileViewProvider{vFile=file:///home/marcin/Dokumenty/NGA/projects/pressio-tutorials/end-to-end-roms/tests/CMakeLists.txt, vFileId=175357, content=com.intellij.psi.AbstractFileViewProvider$PsiFileContent@555a4b68, eventSystemEnabled=true}]; node.length=1004; doc.text!=file.text; file name:CMakeLists.txt; type:com.cmakeplugin.CMakeFileType@33fb5885; lang:Language: CMake [Plugin: artsiomch.cmake]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:83)
	at com.intellij.diagnostic.PluginException.logPluginError(PluginException.java:100)
	at com.intellij.psi.impl.DocumentCommitThread.assertAfterCommit(DocumentCommitThread.java:317)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$doCommit$5(DocumentCommitThread.java:300)
	at com.intellij.psi.impl.PsiDocumentManagerBase.commitToExistingPsi(PsiDocumentManagerBase.java:422)
	at com.intellij.psi.impl.PsiDocumentManagerBase.lambda$finishCommitInWriteAction$5(PsiDocumentManagerBase.java:396)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:238)
	at com.intellij.psi.impl.PsiDocumentManagerBase.lambda$finishCommitInWriteAction$6(PsiDocumentManagerBase.java:391)
	at com.intellij.psi.impl.PsiDocumentManagerBase.executeInsideCommit(PsiDocumentManagerBase.java:507)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommitInWriteAction(PsiDocumentManagerBase.java:389)
	at com.intellij.psi.impl.PsiDocumentManagerImpl.finishCommitInWriteAction(PsiDocumentManagerImpl.java:125)
	at com.intellij.psi.impl.PsiDocumentManagerBase$2.run(PsiDocumentManagerBase.java:356)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1023)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommit(PsiDocumentManagerBase.java:353)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$commitUnderProgress$2(DocumentCommitThread.java:138)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$safeTransferToEdt$6(NonBlockingReadActionImpl.java:596)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:513)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:75)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:118)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:42)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:918)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:766)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:450)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:449)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:624)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:447)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:493)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

# Other Information
```
=== About ===
Build version: PyCharm 2022.2.2 Build: #PY-222.4167.33 September 15, 2022
Theme: Darcula
JRE: 17.0.4+7-b469.53, JetBrains s.r.o.
JVM: 17.0.4+7-b469.53, OpenJDK 64-Bit Server VM, JetBrains s.r.o.
Operating System: Linux 5.19.0-76051900-generic (amd64)

=== Plugins ===
Custom plugins: [CMake simple highlighter (222.4.0), TeXiFy IDEA (0.7.22), Makefile Language (222.4167.21), Requirements (2022.3-222-EAP-SNAPSHOT)]
Disabled plugins:[]
```

**Comment by ArtsiomCh** [comment]
Thanks for reporting!
Looks like some kind of glitch inside ide's `Document` saving mechanism. Also don't see any of my classes in stacktrace...
Did it happen right after any action provided by my plugin (CMake simple highlighter)?
Was it one time exception or you see it often? Or may be even can reproduce?

**Comment by marcinwrobel1986** [comment]
Hello @ArtsiomCh ,
that happend basically after installation.
I got en error in CMake `message(STATUS "something")` , by error I mean error in syntax and then it crashed.
That was one time occurance.
I can not reproduced that. After restart was ok.
I will reach you when that happend again and I can reproduce that.
I posted the issue, becasue I thought you might find some useful info there.
Thanks and best regards,
Marcin
